src/service/test_case.rs

```diff
@@ -0,0 +1,38 @@
+use crate::entities::{Scope, TestCase};
+use crate::{CsesApi, Filesystem, Resources, Storage, RP};
+use anyhow::{Context, Result};
+
+pub fn get_test_cases(res: &mut Resources<impl RP>, scope: &Scope, task_id: &str) -> Result<()> {
+    let test_cases = fetch_test_cases(res, scope, task_id)?;
+    save_test_cases(res, test_cases)?;
+    Ok(())
+}
+
+fn fetch_test_cases(
+    res: &mut Resources<impl RP>,
+    scope: &Scope,
+    task_id: &str,
+) -> Result<Vec<TestCase>> {
+    (|| -> Result<_> {
+        let token = res.storage.get().get_token();
+        let response = res.api.get_test_case_list(token, &scope, task_id)?;
+        Ok(response.test_cases)
+    })()
+    .context("Failed querying test cases from the server.")
+}
+
+fn save_test_cases(res: &mut Resources<impl RP>, test_cases: Vec<TestCase>) -> Result<()> {
```

Comment:
This should probably also take the output directory as a parameter. (At least currently in the `test-case-fetch-command` it is possible to specify the directory).